src/main/java/net/glowstone/block/blocktype/BlockType.java

```diff
@@ -306,27 +307,16 @@ public final void rightClickBlock(GlowPlayer player, GlowBlock against, BlockFac
         }
 
         if (getMaterial().isSolid()) {
-            GlowChunk[] chunks = new GlowChunk[]{
-                    target.getChunk(),
-                    target.getWorld().getChunkAt(target.getChunk().getX() + 1, target.getChunk().getZ()),
-                    target.getWorld().getChunkAt(target.getChunk().getX() + 1, target.getChunk().getZ() + 1),
-                    target.getWorld().getChunkAt(target.getChunk().getX(), target.getChunk().getZ() + 1),
-                    target.getWorld().getChunkAt(target.getChunk().getX() - 1, target.getChunk().getZ()),
-                    target.getWorld().getChunkAt(target.getChunk().getX() - 1, target.getChunk().getZ() - 1),
-                    target.getWorld().getChunkAt(target.getChunk().getX() - 1, target.getChunk().getZ() + 1),
-                    target.getWorld().getChunkAt(target.getChunk().getX(), target.getChunk().getZ() - 1),
-                    target.getWorld().getChunkAt(target.getChunk().getX() + 1, target.getChunk().getZ() - 1),
-            };
             BlockBoundingBox box = new BlockBoundingBox(target);
-            for (GlowChunk chunk : chunks) {
-                for (Entity e : chunk.getEntities()) {
-                    GlowEntity entity = (GlowEntity) e;
-                    if (!(entity instanceof LivingEntity)) {
-                        continue;
-                    }
-                    if (entity.intersects(box)) {
-                        return;
-                    }
+            BoundingBox searchBox = BoundingBox.copyOf(box);
+            Vector vec = new Vector(32, 32, 32);
+            searchBox.minCorner.subtract(vec);
+            searchBox.maxCorner.add(vec);
+            List<Entity> entities = target.getWorld().getEntityManager().getEntitiesInside(searchBox, null);
+            for (Entity e : entities) {
+                GlowEntity entity = (GlowEntity) e;
+                if (entity instanceof LivingEntity && entity.intersects(box)) {
```

Comment:
`entity.intersects(box)` is repetitive because `getEntitiesInside()` already checks this.